Remotion/ObjectBinding/Web.ClientScript/Scripts/BocList.ts

```diff
@@ -0,0 +1,711 @@
+    var realTableHeadRowChildren = realTableHeadRow.children();
+  
+    var fakeTableHeadContainer = scrollableContainer.parent().children('div.bocListFakeTableHead').first();
+    var fakeTableHead = fakeTableHeadContainer.find('thead');
+    var fakeTableHeadRow = fakeTableHead.children().eq(0);
+    var fakeTableHeadRowChildren = fakeTableHeadRow.children();
+  
+    // store cell widths in array
+    var realTableHeadCellWidths = new Array();
+    realTableHeadRowChildren.each(function (this: HTMLElement, index)
+    {
+      realTableHeadCellWidths[index] = $(this).width();
+    });
+  
+    // apply widths to fake header
+    fakeTableHeadRowChildren.width(function (index, itemWidth)
+    {
+      var width = realTableHeadCellWidths[index];
+      return width;
+    });
+  
+    var fakeTableHeadWidthContainer = fakeTableHeadContainer.children ('div').first();
+    fakeTableHeadWidthContainer.width(realTableWidth);
+    var fakeTableHeadContainerHeight = fakeTableHeadContainer.height();
+    scrollableContainer.css({ top: fakeTableHeadContainerHeight});
+    realTable.css({ 'margin-top': fakeTableHeadContainerHeight * -1 });
+  
+    fakeTableHeadContainer.show();
+  }
+
+  private static FixHeaderPosition(tableContainer: JQuery, scrollableContainerJQuery: JQuery): void
+  {
+    var scrollableContainer = scrollableContainerJQuery[0];
+    var fakeTableHeadContainer = tableContainer.children('div.bocListFakeTableHead').first()[0];
+    var scrollLeft = scrollableContainer.scrollLeft;
+    var previousScrollLeft = scrollableContainerJQuery.data("bocListPreviousScrollLeft");
+    var fakeTableHeadScrollLeft = fakeTableHeadContainer.scrollLeft;
+
+    var hasScrollMoveFromScrollbar = previousScrollLeft !== scrollLeft;
+    var hasScrollMoveFromColumnHeader = previousScrollLeft !== fakeTableHeadScrollLeft;
+    if (!hasScrollMoveFromScrollbar && !hasScrollMoveFromColumnHeader)
+      return;
+
+    scrollLeft = hasScrollMoveFromColumnHeader ? fakeTableHeadScrollLeft : scrollLeft;
+    scrollableContainerJQuery.data("bocListPreviousScrollLeft", scrollLeft);
+
+    if (hasScrollMoveFromColumnHeader)
+      scrollableContainer.scrollLeft = scrollLeft;
+
+    if (hasScrollMoveFromScrollbar)
+      fakeTableHeadContainer.scrollLeft = scrollLeft;
+  }
+
+  public static InitializeNavigationBlock (pageNumberField: JQuery, pageIndexField: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsObject ('pageNumberField', pageNumberField);
+    ArgumentUtility.CheckNotNullAndTypeIsObject('pageIndexField', pageIndexField);
+  
+    pageNumberField.bind('change', function () {
+  
+      var pageNumber = parseInt(pageNumberField.val(), 10);
+      if (isNaN (pageNumber) || !TypeUtility.IsInteger (pageNumber))
+      {
+        if (pageNumberField.val().length > 0)
+          setTimeout(function () { pageNumberField.focus(); }, 0);
+  
+        pageNumberField.val("" + (parseInt(pageIndexField.val(), 10) + 1));
```

Comment:
In a previous PR, we had a cast to number instead of adding a blank string. Maybe, that was just because the other code was jquery. There is also a ticket for improving these things. Please add a TODO. (@a-ctor Ticket please)

Reply:
This code is also jQuery but the case your are mentioning was `.css(name, value)` which had an overload for number. `.val` does not have an overload for number (at least the typings do not have one, jQuery converts any numbers to strings). So in this case we are stuck with converting to a `string` or adjusting the typings to add an overload with `number`.

Reply:
Thanks. probably should do this consistently in the cleanup-PX task after the jquery migration

Reply:
@MichaelKetting So change to `'' + ` and create a cleanup issue after the jQuery migration?

Reply:
The clenaup issue we already have. Change the quotes now.

Reply:
What's the number of the clean up issue? I can't find it anywhere

Reply:
For continuity, it's [RM-7699](https://re-motion.atlassian.net/browse/RM-7699)